contracts/colony/IColony.sol

```diff
@@ -247,24 +247,44 @@ interface IColony is ColonyDataTypes, IRecovery {
   /// @notice Install an extension to the colony. Secured function to authorised members.
   /// @param extensionId keccak256 hash of the extension name, used as an indentifier
   /// @param version The new extension version to install
-  function installExtension(bytes32 extensionId, uint256 version) external;
+  /// @return extension The address of the extension installation
+  function installExtension(bytes32 extensionId, uint256 version) external returns (address extension);
 
+  /// @dev DEPRECATED
   /// @notice Upgrade an extension in a colony. Secured function to authorised members.
   /// @param extensionId keccak256 hash of the extension name, used as an indentifier
   /// @param newVersion The version to upgrade to (must be one larger than the current version)
   function upgradeExtension(bytes32 extensionId, uint256 newVersion) external;
```

Comment:
That's a fair point... keeping the versions on the network is probably right so it can play nicely with non-upgraded colonies, but once the colony is upgraded they don't really make sense anymore since as you say they don't do anything -- anyone using the contracts will need to migrate over to the new interface, so adding support for the old functions is perhaps not necessary. I'm happy to throw in a `migrate` function instead.